### PR TITLE
Refactor/move search fetch

### LINKED ISSUE #2: build out the app component

### PR DESCRIPTION
**Pull request by orlov-n**
What does this pull request add?

Everything.

Which issues does it close?

closes #2  as well as many others

### CODE REVIEW

File: cypress/e2e/spec.cy.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+describe('empty spec', () => {
```
Comment (irmakerdem) [inline cypress/e2e/spec.cy.js:1]: This file should be deleted or edited to make it specific to your project.

File: cypress/fixtures/example.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (irmakerdem) [inline cypress/fixtures/example.json:1]: This file should be deleted or edited to make it specific to your project.

File: src/App/App.css [hunk 3]
```diff
@@ -0,0 +1,151 @@
+/* * {
+  border: rgb(141, 141, 242) solid 1.5px;
+  background-color: #e9dfcc;
+} */
+
+.App {
+  margin-left: auto;
+  margin-right: auto;
+  min-width: 320px;
```
Comment (irmakerdem) [inline src/App/App.css:9]: You could use a px to em converter to make all of your styling responsive: https://www.w3schools.com/tags/ref_pxtoemconversion.asp

File: src/App/App.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+const App = () => {
+  const [bookList, setBookList] = useState({});
+  const [userInput, setUserInput] = useState("");
+  const [userSearchResults, setUserSearchResults] = useState([]);
+
+  useEffect(() => {
+    getBookList().then((response) => {
+      const acceptableFormats = response.results.filter((item) => {
+        if (item.formats["text/html"] !== undefined) {
```
Comment (irmakerdem) [inline src/App/App.js:19]: `!== undefined` is truthy. Could you use the bang operator instead for this conditional?

File: src/App/App.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+      <Route exact path="/" render={() => <WelcomePage />} />
+
+      {bookList.length > 0 && (
+        <Route
+          exact
+          path="/book/:book_id"
+          render={(match) => {
+            return (
+              <FullBook
+                bookList={userInput ? userSearchResults : bookList}
```
Comment (irmakerdem) [inline src/App/App.js:56]: Nice ternary! 😎

File: src/BookCard/BookCard.css [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+  transform-origin: left;
+  transform: rotateY(0deg);
+  opacity: 1;
+  transition: all 5s ease-in-out;
+  animation: turnpage 20s infinite linear;
+  overflow: hidden;
+
+}
+
+.pagenext {
```
Comment (irmakerdem) [inline src/BookCard/BookCard.css:28]: Make sure to use kebab case for class names 🍢

File: src/FullBook/FullBook.css [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  /* vertical-align: middle; */
+  position: relative;
+  left: 50%;
+  transform: translateX(-50%);
+  margin-top: 14rem;
+  margin-bottom: 11rem;
+  /* height: 75rem; */
+}
+
+embed {
```
Comment (irmakerdem) [inline src/FullBook/FullBook.css:14]: Whoa, what is `embed`?

File: src/SearchBar/SearchBar.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import React, { useState } from "react";
+import "./SearchBar.css";
+import { NavLink } from "react-router-dom";
+
+export const SearchBar = ({ handleSearch }) => {
+  const [query, setQuery] = useState("");
```
Comment (irmakerdem) [inline src/SearchBar/SearchBar.js:6]: Nice 🪝!

File: src/apiCalls.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+const getBook = (bookId) => {
+  return fetch(`https://gutendex.com/books/${bookId}`).then((response) => {
+    if (response.ok) {
+      return response.json();
+    }
+    throw Error(response.statusText);
+  });
+};
+
+export { getBookList, searchQuery, getBook };
```
Comment (irmakerdem) [inline src/apiCalls.js:31]: Wow you have 3 fetch calls!! 🙀

**Comment by irmakerdem** [comment]
Looks good!

File: cypress/e2e/spec.cy.js (hunk 1 above)
Comment (orlov-n) [reply cypress/e2e/spec.cy.js:1]: Thanks.

File: cypress/fixtures/example.json (hunk 2 above)
Comment (orlov-n) [reply cypress/fixtures/example.json:1]: Thanks.

File: src/App/App.css (hunk 3 above)
Comment (orlov-n) [reply src/App/App.css:9]: This particular min-width is in pixels to limit the minimum amount of width using the smallest size phone screen as a hard stop.  Thanks for that other resource though.  Will look into it for general use.

File: src/App/App.js (hunk 4 above)
Comment (orlov-n) [reply src/App/App.js:19]: I thought there was some rule about using bang operators with object's properties that make it not work or something?  Need to do further research.

File: src/App/App.js (hunk 5 above)
Comment (orlov-n) [reply src/App/App.js:56]: Thanks.  This one solved a lot of headache!

File: src/BookCard/BookCard.css (hunk 6 above)
Comment (orlov-n) [reply src/BookCard/BookCard.css:28]: That's a great point.  Will need to refactor.

File: src/FullBook/FullBook.css (hunk 7 above)
Comment (orlov-n) [reply src/FullBook/FullBook.css:14]: That's for rendering actual files and other odd things in your window.  You can also use iframe.

File: src/apiCalls.js (hunk 9 above)
Comment (orlov-n) [reply src/apiCalls.js:31]: This project ended up being a lot more complex than I had planned  :)

**Pull request merged**